File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/utils/PackageManagerUtils.java
package com.zhongjh.albumcamerarecorder.utils;

import android.content.pm.FeatureInfo;
import android.content.pm.PackageManager;

/**
 * 有关手机模块工具
 *
 * @author zhongjh
 * @date 2019/3/25
 */
public class PackageManagerUtils {

    /**
     * 判断是否支持闪光灯
     * @param pm PackageManager
     * @return 是否
     */
    public static boolean isSupportCameraLedFlash(PackageManager pm) {
        if (pm != null) {
            FeatureInfo[] features = pm.getSystemAvailableFeatures();
            if (features != null) {
                for (FeatureInfo ignored : features) {
                    return true;
                }
            }
        }
        return false;
    }

}

File: multilibrary/src/main/java/com/zhongjh/albumcamerarecorder/utils/PackageManagerUtils.kt
package com.zhongjh.albumcamerarecorder.utils

import android.content.pm.PackageManager

/**
 * 有关手机模块工具
 *
 * @author zhongjh
 * @date 2019/3/25
 */
object PackageManagerUtils {
    /**
     * 判断是否支持闪光灯
     * @param pm PackageManager
     * @return 是否
     */
    @JvmStatic
    fun isSupportCameraLedFlash(pm: PackageManager?): Boolean {
        if (pm != null) {
            val features = pm.systemAvailableFeatures
            for (ignored in features) {
                return true
            }
        }
        return false
    }
}